src/api/controllers/db-controller.ts

```diff
@@ -804,11 +911,18 @@ export async function getTxFromDataStore(
     dbTx = txQuery.result;
   }
 
-  const parsedTx = parseDbTx(dbTx);
+  let parsedTx = parseDbTx(dbTx);
 
   // If tx type is contract-call then fetch additional contract ABI details for a richer response
   if (parsedTx.tx_type === 'contract_call') {
-    await getContractCallMetadata(db, dbTx, parsedTx);
+    const transaction = await getContractCallMetadata(
+      db,
+      parseDbTx(dbTx) as ContractCallTransaction,
+      dbTx
+    );
+    if (transaction) {
+      parsedTx = transaction as ContractCallTransaction;
+    }
```

Comment:
Same here, reuse the new function above.

Reply:
`getTxFromDataStore` function is called from various APIs outside `db-controller.ts`. I didn't replicate the entire implementation in `getTxsFromDataStore` and just refactored the code needed to work for this PR.